rfc/ipni-w3c.md

```diff
@@ -146,67 +145,93 @@ flowchart TD
 ```
 
 ## Proposal
-Provide a ipni/offer UCAN ability to submit the location of a CAR and a CARx-index that a user has stored with w3s, to make the them discoverable via IPFS implementations and other IPNI consumers.
+
+Provide a ipni/offer invocation UCAN to request that w3 publish to IPNI an advertisement that indexes the multihashes in the user-created inclusion claims for looking up the content claims bundle. This data represented by these multihashes will then be discoverable via IPFS implementations and other IPNI consumers.
 
 The ipni/offer is optional, and if not invoked, then the user's data is not indexed.
 
-If opting-in to IPNI indexing, the CAR file and the CARx-index are created by the client. The client may choose which multihashes they want indexed and specify this in the ipni/offer request.
+If opting-in to IPNI indexing, a request in to form of an invocation UCAN is created by the client and refers to the content claims bundle that was created when the client stored a data blob. The client may choose which multihashes they want indexed and specify this in the request. By default all multihashes in the inclusion claim are indexed if opting to index data.
 
 ### Indexing stored data
 
-```mermaid
+``mermaid
 sequenceDiagram
     actor Alice
-    Alice->>w3s: put CAR + CARx-index
+    Alice->>w3s: put blob + inclusion claim
     activate w3s
     w3s-->>w3s: create location claims
-    w3s-->>w3s: create/store claims bucket
-    w3s-->>Alice: location claims
+    w3s-->>w3s: create/store claims bundle
+    w3s-->>Alice: claims bundle ID and URL
     deactivate w3s
-    activate Alice
-    Alice->>w3s: ipni/offer (inclucion, locations CIDs)
-    deactivate Alice
+    Alice->>w3s: get blob status
+    activate w3s
+    w3s-->>Alice: ok, claims bundle CID
+    deactivate w3s
+    Alice->>w3s: ipni/offer (claims-CID, opt: mh set))
     activate w3s
-    w3s-->>w3s: fetch & verify CARx-index
+    w3s-->>w3s: fetch & verify claims bundle
     w3s-->>w3s: write advertisement
-    w3s-)ipni: publish head (CID)
+    w3s-)ipni: announce head (CID)
     deactivate w3s
     ipni-->>w3s: fetch advert
     activate ipni
     ipni-->>ipni: index entries
     deactivate ipni
 ```
+- User creates their data blob and inclusion claim, then uploads these to w3s to the pre-signed PUT URL.
+  - The inclusion claim listing sub-blocks within the blob is created by the client whether or not they intend to have their data indexed. This is because these blocks may still be needed to find data within the blob, and to allow indexing to be done at a later point without having the re-read the entire blob.
+  - W3s does not need to immediately verify the correctness of the inclusion claim, but can at any time and then replace the inclusion claim with its own to certify that it was verified.
```

Comment:
OK, that's fine. If the verified w3s claim is there, then it seemed like there was no reason to keep the user-created claim. Also, if w3s moves the blob to a new location later on then the user's claim will no longer be correct.

Reply:
That assumes that our claims are superior, but in reality it’s just more trustworthy from our own perspective. On the other hand others may trust claim issuer more than they trust us. Does that make sense ? I don’t think a our claims as replacement for user claims they are just ours so we can trust them.

> Also, if w3s moves the blob to a new location later on then the user's claim will no longer be correct.

We should not move for the duration of claim we issued or we aren’t upholding our commitment. That said, that is also why I’m somewhat biased for having location claims separate form inclusion ones, because that way later is agnostic of former